### ISSUE TITLE
Add time conversion functions

### ISSUE THREAD

**Issue by aswan** [brimdata/zed#808]
Following up from #789, this issue is to introduce methods for writing timestamps with nanosecond precision that aren't susceptible to floating point rounding bugs.

We could start with some functions that convert from strings or large integers, e.g.:
`ts = Time.fromISO("2020-05-20T12:00:00.000Z")`, or `ts = Time.fromNanoseconds(1589976000000000000)`, or ts = Time.fromMicroseconds(1589976000000000)`

@mccanne also proposed that we could push this handling all the way into the grammar so that something like `Time.fromFloat(1589976000000000.123456789)` would generate an AST that, instead of having a float literal AST node, would just include the operand as a string in the AST so it could be parsed by the existing `nano.Parse()`.

**Comment by aswan** [comment]
#822 has the first part of this.  The conversion from floats with special treatment in the parser needs to be fleshed out a bit more so leaving this open and unassigned for somebody to pick up that work.

**Comment by philrz** [comment]
As discussed in #2334, the time conversion functions were backed out since an approach based on casting is now used instead. ISO timestamps are now supported in ZSON and lack the same exposure to floating point rounding problems. As also disclosed there, there's currently no plans to take special steps to avoid floating point rounding problems, e.g., this can still occur as of current Zed commit 620f282:

```
$ zq -version
Version: v0.30.0-25-g620f282c

$ echo '{"mytime": 1617490638.707690}' | zq 'timeval:=time(mytime)' -
{mytime:1.61749063870769e+09,timeval:2021-04-03T22:57:18.707689984Z}
```

For the case of Zeek TSV logs that represent timestamps in floating point numbers, but mark them as being of a specific `time` type, Zed's Zeek TSV reader is capable of converting these to the corresponding Zed `time` type without the same effect shown above.

```
$ cat mytime.tsv 
#separator \x09
#set_separator	,
#empty_field	(empty)
#unset_field	-
#fields	timeval
#types	time
1617490638.707690

$ zq -z mytime.tsv 
{timeval:2021-04-03T22:57:18.70769Z}
```

In conclusion, if users import data that uses the increasingly-popular ISO timestamp formats, or use Zeek TSV logs (that represents one of the common log formats we encounter that still outputs such values as floats), they should not be exposed to floating point round problems when working with the ultimate Zed `time` values. If other float-to-`time` conversion requirements surface among users, we could revisit an approach like the one proposed earlier in this issue. But for now it seems this issue has served its original purpose and can be closed.
